scripts/Diffraction/isis_powder/hrpd.py

```diff
@@ -235,71 +235,37 @@ def _subtract_prompt_pulses(self, ws, tof_res=0.002):
-                )
-                comp_func[0].setAttributeValue("CentreShift", cen)
-                comp_func[0].setHeight(ws.readY(ispec)[ws.yIndexOfX(cen)])
-                comp_func[0].addConstraints(f"{-15}<Centre<{15}")
-                comp_func[0].addConstraints(f"{0}<Intensity")
-                comp_func[1]["A0"] = min(ws.readY(ispec)[ws.yIndexOfX(xlo)], ws.readY(ispec)[ws.yIndexOfX(xhi)])
+                comp_func = self._setup_single_prompt_pulse_function(ws, ispec, cen, xlo, xhi)
                 func.add(comp_func)
                 func.setDomainIndex(ipulse, ipulse)
                 key_suffix = f"_{ipulse}" if ipulse > 0 else ""
                 fit_kwargs["InputWorkspace" + key_suffix] = ws.name()
                 fit_kwargs["StartX" + key_suffix] = xlo
                 fit_kwargs["EndX" + key_suffix] = xhi
                 fit_kwargs["WorkspaceIndex" + key_suffix] = ispec
-                # work out how to exclude
-                exclude = []
-                diff_consts = si.diffractometerConstants(ispec)
-                for dpk in dpks_bragg:
-                    tof_pk = UnitConversion.run("dSpacing", "TOF", dpk, 0, DeltaEModeType.Elastic, diff_consts)
-                    tof_pk_lo = tof_pk * (1 - tof_res)
-                    tof_pk_hi = tof_pk * (1 + tof_res)
-                    if xlo < tof_pk_lo < xhi or xlo < tof_pk_hi < xhi or (tof_pk_lo < xlo and tof_pk_hi > xhi):
-                        exclude.extend([tof_pk_lo, tof_pk_hi])
+                # exclude x-ranges where Bragg peaks overlap prompt pulse within some fractional resolution
+                exclude = self._get_tofs_to_exclude(si, ispec, dpks_bragg, xlo, xhi)
                 if exclude:
                     func.fixParameter(f"f{ipulse}.f1.A0")  # fix constant background - Bragg peak can mess with bg
                     fit_kwargs["Exclude" + key_suffix] = exclude
             # check that at least one fit region has no overlapping prompt pulse
             if len([key for key in fit_kwargs.keys() if "Exclude" in key]) < len(npulses):
-                # tie peak parameters to be common for all prompt pulses
-                for idomain in range(len(npulses) - 1):
-                    for param_name in ["Intensity", "Sigma", "Exponent", "Skew", "Centre"]:
-                        func.tie(f"f{idomain}.f0.{param_name}", f"f{len(npulses) - 1}.f0.{param_name}")  # tie to first
-                # fix some peak parameters
-                for param_name in ["Sigma", "Exponent", "Skew"]:
-                    func.fixParameter(f"f{len(npulses) - 1}.f0.{param_name}")
+                self._add_ties_to_multidomain_prompt_pulse_func(func)
                 fit_output = mantid.Fit(Function=func, **fit_kwargs)
                 # subtract prompt pulse only from spectrum
                 success = fit_output.OutputStatus == "success" or "Changes in function value are too small" in fit_output.OutputStatus
```

Comment:
Here it is also a bit confusing expression
```suggestion
                is_success_status = fit_output.OutputStatus == "success"
                is_small_change = "Changes in function value are too small" in fit_output.OutputStatus

                success = is_success_status or is_small_change
```